feat: use current sass package to parse scss files

miluoshi opened the pull request:
There's no need for custom parser for `*.scss` files. `sass` package can handle `scss` syntax. This PR updates `sass` to the latest version and replaces deprecated `sass.renderSync` -> `sass.compile`.

Related: #632

rumpl commented:
I <3 our dogfooding, it looks like the build is failing because `query-ast` is unused now :)

miluoshi commented:
Nice :) `query-ast` has been removed now as well.

rumpl commented:
Just kicked the CI, we now have some tests failing, could you take a look?

miluoshi commented:
I've updated the PR. `@import`/`@use` parsing was not done correctly in some cases. 

- I needed to update tested scss files for them to be valid (e.g. `@use` rules need to be always before `@import`).
- `sass` compiler is not reporting usage of its' internal modules (e.g. `@use "sass:math"`), so I've added regex matching against every file. It is not an optimal solution performance-wise. 
 
Imho a better solution to above performance problem would be that `sass` package would be reported as "used" by depcheck automatically if any `*.{scss,sass}` file is parsed, but then the following parts of spec file would have to be updated to not fail test:
```js
{
  name: 'support SASS/SCSS syntax',
  expected: {
    missing: {
      / ...
      sass: ['scss2.scss'], // <-- this should include each parsed sass file
    },
    using: {
      // ...
      sass: ['scss2.scss'], // <-- this should include each parsed sass file
    }
  },
```

miluoshi commented:
CI failed on one machine. Did it fail because of this error?

```
> cross-env NODE_ENV=test nyc mocha ./test ./test/special --timeout 20000 && nyc report --reporter=text-lcov > ./coverage/coverage.lcov

'cross-env' is not recognized as an internal or external command,
operable program or batch file.
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! depcheck@0.0.1 test-coverage: `cross-env NODE_ENV=test nyc mocha ./test ./test/special --timeout 20000 && nyc report --reporter=text-lcov > ./coverage/coverage.lcov`
npm ERR! Exit status 1
npm ERR! 
npm ERR! Failed at the depcheck@0.0.1 test-coverage script.
```

I'm not sure what are problems with codecov, because I can't see the checks.

Anyway, I've also removed `node-sass` in the latest commit, because [it was not being used for some time](https://github.com/depcheck/depcheck/commit/04217610daf58b02509d75e04e589e1666b5fd6c)...

miluoshi commented:
Thank you. I'll check and fix the tests

rumpl commented:
Thanks! Don’t mind the codecov, it’s kinda broken and I think I’ll remove it at some point

miluoshi commented:
I can see it is failing only in Windows machine, but I have no idea why. I currently don't have access to a Windows machine to debug it. All the tests pass in Linux/MacOS.

I can get to Windows machine later this week to see what might be causing it.
